src/utils/files/download.ts

```diff
@@ -0,0 +1,17 @@
+export function downloadString(
+  content: string,
+  filename = 'file.txt',
+  mimeType = 'text/plain'
+) {
+  const blob = new Blob([content], { type: `${mimeType};charset=utf-8;` });
+  const url = URL.createObjectURL(blob);
+
+  const link = document.createElement('a');
```

Comment:
Even thought these DOM manipulations formally happen outside the React component, there is another approach, that's closer to the course goal - return url from this function and use it via ref to the anchor component inside the react component.